lib/manager/poetry/readme.md

```diff
@@ -0,0 +1,170 @@
+## Overview
+
+#### Name of package manager
+
+[Poetry](https://poetry.eustace.io/)
+
+---
+
+#### What language does this support?
+
+Python
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+pip
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+Yes
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+TOML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+TOML can be parsed line by line
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+No
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+TODO
```

Comment:
There are several types

- "classic"  `numpy="1.16.1"` or `numpy={ version="1.16.1" }`
- git `cleo = { git = "https://github.com/sdispater/cleo.git", branch = "master" }`
- path, target can be directory or file .py `docs = { path = "./packages/docs" }` 
- file, target can be archive or wheel `docs = { file = "./my_wheel.whl" }` 

Not totally sure about file and path though